cluster/ci/monitoring/mixins/prometheus_out/alertmanager.yaml

```diff
@@ -1,31 +1,30 @@
 global:
   resolve_timeout: 5m
-
-route:
-  group_by: ['alertname', 'job']
-  group_wait: 30s
-  group_interval: 5m
-  repeat_interval: 2h
-  receiver: 'slack-notifications'
-  routes:
-  - receiver: 'slack-build-cop'
-    match:
-      team: build-cop
-
 receivers:
-- name: 'slack-notifications'
+- name: slack-notifications
   slack_configs:
-  - channel: '#ops-testplatform'
-    api_url: '{{ api_url }}'
+  - api_url: '{{ api_url }}'
+    channel: '#ops-testplatform'
     icon_url: https://avatars3.githubusercontent.com/u/3380462
     text: '{{ template "custom_slack_text" . }}'
-- name: 'slack-build-cop'
+- name: slack-build-cop
   slack_configs:
-  - channel: '#build-cop-alerts'
-    api_url: '{{ api_url }}'
+  - api_url: '{{ api_url }}'
+    channel: '#build-cop-alerts'
     icon_url: https://avatars3.githubusercontent.com/u/3380462
-    text: '{{ template "custom_slack_text" . }}'
     link_names: true
-
+    text: '{{ template "custom_slack_text" . }}'
+route:
+  group_by:
+  - alertname
+  - job
+  group_interval: 5m
+  group_wait: 30s
+  receiver: slack-notifications
+  repeat_interval: 2h
+  routes:
+  - match:
+      team: build-cop
+    receiver: slack-build-cop
```

Comment:
This part needs to be automated too.

Reply:
Is this still WIP?

Reply:
> Is this still WIP?

WIP here is the automation part.
Before `./alertmanager.yaml` is hard-coding.
Now `./mixins/prometheus_out/alertmanager.yaml` is generated by jsonnet.

The 1st commit is to do the transaction above, but need to make the existing 2 channels work (the target is jsonnet should not change any existing `yamls` and `jsons`).

The 2nd gives an example of adding a channel and using it for some job failure.